src/storage/storage.cc

```diff
@@ -705,13 +705,13 @@ rocksdb::Status Storage::writeToDB(engine::Context &ctx, const rocksdb::WriteOpt
     updates->PutLogData(ServerLogData(kReplIdLog, replid_).Encode());
   }
 
-  if (ctx.is_txn_mode) {
-    if (ctx.batch == nullptr) {
-      ctx.batch = std::make_unique<rocksdb::WriteBatchWithIndex>();
+  if (ctx.txn_context_enabled) {
+    if (ctx.batch != nullptr) {
+      WriteBatchIndexer handle(ctx);
+      auto s = updates->Iterate(&handle);
+      if (!s.ok()) return s;
+      ctx.batch = nullptr;
```

Comment:
Why we need to clear ctx.batch if it's not null?

Reply:
Got it.

And why do we now avoid to create a new write batch if ctx.batch is null?

Reply:
Ahh I think no actually write happens so we can omit it.